/**:
  ros__parameters:

    # Topics
    pointCloudTopic: "/points_raw"           # Point cloud data
    imuTopic: "/imu"                            # IMU data
    odomTopic: "odometry/imu"                   # IMU pre-preintegration odometry, same frequency as IMU
    gpsTopic: "/gps/fix"                        # GPS odometry topic from navsat, see module_navsat.launch file
    #gpsOdomTopic: "/NavOdom"                    # GPS odometry topic from navsat, see module_navsat.launch file
    gpsOdomTopic: "/odometry/filtered" #"/lio_sam/gps/odom"

    # Frames
    lidarFrame: "os_sensor"
    baselinkFrame: "base_link"
    odometryFrame: "odom"
    mapFrame: "map"

    # GPS Settings
    useImuHeadingInitialization: true           # if using GPS data, set to "true"
    useGpsElevation: true                       # if GPS elevation is bad, set to "false"
    gpsCovThreshold: 2.0                         # m^2, threshold for using GPS data
    poseCovThreshold: 25.0                       # m^2, threshold for using GPS data
    gpsPointThreshold: 1.0                       # m, distance between GPS points for adding factor

    # Export settings
    savePCD: true                               # https://github.com/TixiaoShan/LIO-SAM/issues/3
    savePCDDirectory: "/Downloads/LOAM/"         # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

    # Sensor Settings
    sensor: ouster                               # lidar sensor type, either 'velodyne', 'ouster' or 'livox'
    N_SCAN: 128                                   # number of lidar channels (i.e., Velodyne/Ouster: 16, 32, 64, 128, Livox Horizon: 6)
    Horizon_SCAN: 2048                            # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048, Livox Horizon: 4000)
    downsampleRate: 1                            # default: 1. Downsample your data if too many
    # points. i.e., 16 = 64 / 4, 16 = 16 / 1
    lidarMinRange: 1.0                           # default: 1.0, minimum lidar range to be used
    lidarMaxRange: 50.0                     # default: 1000.0, maximum lidar range to be used

    # IMU Settings
    imuAccNoise: 0.01 # Spatial - 0.062, Certus - 0.003
    imuGyrNoise: 0.001 # Spatial - 0.0067, Certus - 0.0002
    imuAccBiasN: 0.00001
    imuGyrBiasN: 0.00001

    imuGravity: 9.80511
    imuRPYWeight: 0.01

    # spatial
    # extrinsicTrans:  [ -0.023,  0.13,  -0.07 ]
    
    # extrinsicRot:    [-1.0,  0.0,  0.0,
    #                    0.0,  1.0,  0.0,
    #                    0.0,  0.0, -1.0 ]
    
    # extrinsicRPY: [ 1.0,  0.0,  0.0,
    #                  0.0,  1.0,  0.0,
    #                  0.0,  0.0,  1.0 ]

    # Certus Evo
    # extrinsicTrans:  [ 0.0,  0.0,  -0.0946 ]
    # extrinsicRot:    [1.0,  0.0,  0.0,
    #                   0.0,  1.0,  0.0,
    #                   0.0,  0.0,  1.0 ]
    
    # extrinsicRPY: [ 1.0,  0.0,  0.0,
    #                 0.0,  1.0,  0.0,
    #                 0.0,  0.0,  1.0 ]

    # Certus
    extrinsicTrans:  [ 0.0,  0.0,  -0.0899]
    extrinsicRot:    [1.0,  0.0,  0.0,
                      0.0,  1.0,  0.0,
                      0.0,  0.0,  1.0 ]
    
    extrinsicRPY: [ 1.0,  0.0,  0.0,
                    0.0,  1.0,  0.0,
                    0.0,  0.0,  1.0 ]

    # LOAM feature threshold
    edgeThreshold: 0.25  # 0.03 PCL curv
    surfThreshold: 0.01  # 0.004 PCL curv
    edgeFeatureMinValidNum: 10
    surfFeatureMinValidNum: 50

    # voxel filter paprams
    odometrySurfLeafSize: 0.2                     # default: 0.4 - outdoor, 0.2 - indoor
    mappingCornerLeafSize: 0.1                    # default: 0.2 - outdoor, 0.1 - indoor
    mappingSurfLeafSize: 0.2                      # default: 0.4 - outdoor, 0.2 - indoor

    # robot motion constraint (in case you are using a 2D robot)
    z_tolerance: 5.0 #5.0                          # meters
    rotation_tolerance: 1000.0 #6.5                   # radians

    # CPU Params
    numberOfCores: 10                            # number of cores for mapping optimization
    mappingProcessInterval: 0.1               # seconds, regulate mapping frequency

    # Surrounding map
    surroundingkeyframeAddingDistThreshold: 0.5   # meters, regulate keyframe adding threshold
    surroundingkeyframeAddingAngleThreshold: 0.1  # radians, regulate keyframe adding threshold
    surroundingKeyframeDensity: 0.4               # meters, downsample surrounding keyframe poses   
    surroundingKeyframeSearchRadius: 25.0         # meters, within n meters scan-to-map optimization
    # (when loop closure disabled)

    # Loop closure
    loopClosureEnableFlag: true
    loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
    surroundingKeyframeSize: 50                   # submap size (when loop closure enabled)
    historyKeyframeSearchRadius: 25.0             # meters, key frame that is within n meters from
    # current pose will be considerd for loop closure
    historyKeyframeSearchTimeDiff: 30.0          # seconds, key frame that is n seconds older will be
    # considered for loop closure
    historyKeyframeSearchNum: 25                  # number of hostory key frames will be fused into a
    # submap for loop closure
    historyKeyframeFitnessScore: 0.3              # icp threshold, the smaller the better alignment

    # Visualization
    globalMapVisualizationSearchRadius: 1000.0    # meters, global map visualization radius
    globalMapVisualizationPoseDensity: 1.0        # meters, global map visualization keyframe density
    globalMapVisualizationLeafSize: 0.1           # meters, global map visualization cloud density
    globalMapLeafSize: 0.1                        # meters, global map cloud density

    # Exclusion box in the Ouster coordinate Frame
    exclusionBox: [ 0.0,  7.0,
                    -1.5,  0.0,
                    -3.0,  5.0  ]